clearnode/channel_service.go

```diff
@@ -103,76 +103,110 @@ func (s *ChannelService) RequestCreate(wallet common.Address, params *CreateChan
-	balance, err := ledger.Balance(NewAccountID(channel.Wallet), asset.Symbol)
-	if err != nil {
-		logger.Error(ErrGetAccountBalance, "error", err)
-		return ChannelOperationResponse{}, RPCErrorf(ErrGetAccountBalance+" for asset %s", asset.Symbol)
-	}
+		// Prevent no-op resize operations
+		if params.ResizeAmount.Cmp(decimal.Zero) == 0 && params.AllocateAmount.Cmp(decimal.Zero) == 0 {
+			return RPCErrorf("resize operation requires non-zero ResizeAmount or AllocateAmount")
+		}
 
-	rawBalance := balance.Shift(int32(asset.Token.Decimals))
-	newChannelRawAmount := channel.RawAmount.Add(*params.AllocateAmount)
+		ledger := GetWalletLedger(tx, common.HexToAddress(channel.Wallet))
+		balance, err := ledger.Balance(NewAccountID(channel.Wallet), asset.Symbol)
+		if err != nil {
+			logger.Error(ErrGetAccountBalance, "error", err)
+			return RPCErrorf(ErrGetAccountBalance+" for asset %s", asset.Symbol)
+		}
 
-	if rawBalance.Cmp(newChannelRawAmount) < 0 {
-		return ChannelOperationResponse{}, RPCErrorf("insufficient unified balance for channel %s: required %s, available %s", channel.ChannelID, newChannelRawAmount.String(), rawBalance.String())
-	}
-	newChannelRawAmount = newChannelRawAmount.Add(*params.ResizeAmount)
-	if newChannelRawAmount.Cmp(decimal.Zero) < 0 {
-		return ChannelOperationResponse{}, RPCErrorf("new channel amount must be positive: %s", newChannelRawAmount.String())
-	}
+		rawBalance := balance.Shift(int32(asset.Token.Decimals))
+		newChannelRawAmount := channel.RawAmount.Add(*params.AllocateAmount)
 
-	allocations := []nitrolite.Allocation{
-		{
-			Destination: common.HexToAddress(params.FundsDestination),
-			Token:       common.HexToAddress(channel.Token),
-			Amount:      newChannelRawAmount.BigInt(),
-		},
-		{
-			Destination: s.signer.GetAddress(),
-			Token:       common.HexToAddress(channel.Token),
-			Amount:      big.NewInt(0),
-		},
+		if rawBalance.Cmp(newChannelRawAmount) < 0 {
+			return RPCErrorf("insufficient unified balance for channel %s: required %s, available %s", channel.ChannelID, newChannelRawAmount.String(), rawBalance.String())
+		}
+		newChannelRawAmount = newChannelRawAmount.Add(*params.ResizeAmount)
+		if newChannelRawAmount.Cmp(decimal.Zero) < 0 {
+			return RPCErrorf("new channel amount must be positive: %s", newChannelRawAmount.String())
+		}
+
+		channel.Status = ChannelStatusResizing
+
+		if err := tx.Save(channel).Error; err != nil {
+			return RPCErrorf("error saving channel in database: %w", err)
+		}
+
+		if params.ResizeAmount.Cmp(decimal.Zero) < 0 {
+			decimalResizeAmount := rawToDecimal(params.ResizeAmount.BigInt(), asset.Token.Decimals)
+			// Lock funds in the channel account.
+			if err := ledger.Record(NewAccountID(channel.Wallet), asset.Symbol, decimalResizeAmount, nil); err != nil {
+				return err
+			}
+			if err := ledger.Record(NewAccountID(channel.ChannelID), asset.Symbol, decimalResizeAmount.Neg(), nil); err != nil {
```

Comment:
Here we use `ChannelID` as `AccountID`, it won't conflict with other existing functionality, right?
I mean, if an account has some funds allocated in the channel, does this amount live only inside `channels` table or is there some ledger record as well?